barretenberg/cpp/src/barretenberg/vm2/tracegen/execution_trace.cpp

```diff
@@ -779,6 +798,30 @@ void ExecutionTraceBuilder::process_registers(ExecutionOpCode exec_opcode,
         trace.set(REGISTER_MEM_TAG_COLUMNS[i], row, static_cast<uint8_t>(registers[i].get_tag()));
     }
 }
+void ExecutionTraceBuilder::process_get_env_var_opcode(
+    TaggedValue envvar_enum, TaggedValue output, TraceContainer& trace, uint32_t row, bool no_error)
+{
+    assert(envvar_enum.get_tag() == ValueTag::U8);
+    const auto& envvar_spec = GetEnvVarSpec::get_table(envvar_enum.as<uint8_t>());
+
+    assert(!no_error || !envvar_spec.invalid_enum && "execution trace loop says no error, but envvar enum is invalid");
+
+    trace.set(row,
+              { {
+                  { C::execution_sel_should_get_env_var, no_error ? 1 : 0 },
+                  { C::execution_sel_envvar_pi_lookup_col0, envvar_spec.envvar_pi_lookup_col0 ? 1 : 0 },
+                  { C::execution_sel_envvar_pi_lookup_col1, envvar_spec.envvar_pi_lookup_col1 ? 1 : 0 },
+                  { C::execution_envvar_pi_row_idx, envvar_spec.envvar_pi_row_idx },
+                  { C::execution_is_address, envvar_spec.is_address ? 1 : 0 },
```

Comment:
Not sure if it applies (or if the envvar spec is "column aware") but you could consider doing sth like this, just saving the column: https://github.com/AztecProtocol/aztec-packages/blob/next/barretenberg/cpp/src/barretenberg/vm2/tracegen/alu_trace.cpp#L35